### PR TITLE
[RFC][AIC-py] hf summarization model parser

### PR DESCRIPTION
**Pull request by jonathanlastmileai**
[RFC][AIC-py] hf summarization model parser

Very similar to text gen

Test:

```
...
    {
      "name": "test_hf_sum",
      "input": "HMS Duncan was a D-class destroyer ...", # [contents of https://en.wikipedia.org/wiki/HMS_Duncan_(D99)]
      "metadata": {
        "model": {
          "name": "stevhliu/my_awesome_billsum_model",
          "settings": {
            "min_length": 100,
            "max_length": 200,
            "num_beams": 1
          }
        }
      }
    },
...
}
```

```
import asyncio
from aiconfig_extension_hugging_face.local_inference.text_summarization import HuggingFaceTextSummarizationTransformer

from aiconfig import AIConfigRuntime, InferenceOptions, CallbackManager

# Load the aiconfig.

mp = HuggingFaceTextSummarizationTransformer()

AIConfigRuntime.register_model_parser(mp, "stevhliu/my_awesome_billsum_model")

config = AIConfigRuntime.load("/Users/jonathan/Projects/aiconfig/cookbooks/Getting-Started/travel.aiconfig.json")
config.callback_manager = CallbackManager([])


def print_stream(data, _accumulated_data, _index):
    print(data, end="", flush=True)


async def run():
    print("Stream")
    options = InferenceOptions(stream=True, stream_callback=print_stream)
    out = await config.run("test_hf_sum", options=options)
    print("Output:\n", out)

    print("no stream")
    options = InferenceOptions(stream=False)
    out = await config.run("test_hf_sum", options=options)
    print("Output:\n", out)


asyncio.run(run())


# OUT:

Stream
Token indices sequence length is longer than the specified maximum sequence length for this model (2778 > 512). Running this sequence through the model will result in indexing errors
/opt/homebrew/Caskroom/miniconda/base/envs/aiconfig/lib/python3.10/site-packages/transformers/generation/configuration_utils.py:430: UserWarning: `num_beams` is set to 1. However, `early_stopping` is set to `True` -- this flag is only used in beam-based generation modes. You should set `num_beams>1` or unset `early_stopping`.
  warnings.warn(
/opt/homebrew/Caskroom/miniconda/base/envs/aiconfig/lib/python3.10/site-packages/transformers/generation/configuration_utils.py:449: UserWarning: `num_beams` is set to 1. However, `length_penalty` is set to `2.0` -- this flag is only used in beam-based generation modes. You should set `num_beams>1` or unset `length_penalty`.
  warnings.warn(
<pad> escorted the 13th Destroyer Flotilla in the Mediterranean and escorited the carrier Argus to Malta during the war  The ship was recalled home to be converted into an escoter destroyer in late 1942. The vessel was repaired and given a refit at Gibraltar on 16 November, and was sold for scrap later that year. The crew of the ship escores the ship to the Middle East, and the ship is a 'disaster' of the </s>Output:
 [ExecuteResult(output_type='execute_result', execution_count=0, data="<pad> escorted the 13th Destroyer Flotilla in the Mediterranean and escorited the carrier Argus to Malta during the war  The ship was recalled home to be converted into an escoter destroyer in late 1942. The vessel was repaired and given a refit at Gibraltar on 16 November, and was sold for scrap later that year. The crew of the ship escores the ship to the Middle East, and the ship is a 'disaster' of the </s>", mime_type=None, metadata={})]
no stream
Output:
 [ExecuteResult(output_type='execute_result', execution_count=0, data="escorted the 13th Destroyer Flotilla in the Mediterranean and escorited the carrier Argus to Malta during the war . The ship was recalled home to be converted into an escoter destroyer in late 1942. The vessel was repaired and given a refit at Gibraltar on 16 November, and was sold for scrap later that year. The crew of the ship escores the ship to the Middle East, and the ship is a 'disaster' of the .", mime_type=None, metadata={})]

```

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/740).
* #753
* __->__ #740

### CODE REVIEW

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/__init__.py
Comment (rossdanlm) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/__init__.py:5]: We should keep this line. It's used in the `remote_inference` part of this extension

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py
Comment (rossdanlm) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:32]: Nit: I know this is copied from text_generation, but can we link to the API instead?

https://github.com/huggingface/transformers/blob/6eba901d88909b4635b02e284723a71a675ebaa0/src/transformers/pipelines/text2text_generation.py#L217-L228

Feel free to edit the `text_generation.py` file too!

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py
Comment (rossdanlm) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:140]: Do you want to follow up with this in antoher diff? I had that if-statement as a safety check becuase sometimes it's not text outputs (not necessarily for this model parser, just generally for streaming it's not always text format) and didn't think I had time to verify

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py
Comment (rossdanlm) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:242]: Can we call this `text_inputs` to match the name of the pipeline text generation params?

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py
Comment (rossdanlm) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:261]: nice!

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py
Comment (rossdanlm) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:302]: btw this is actually a bug that I fixed in https://github.com/lastmile-ai/aiconfig/pull/739. Can you change this to be `OutputDataWithStringValue` instead? Thanks!

I should also do a more thorough sweep of this bug, but will wait until after you land this to avoid merge conflicts

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/__init__.py
Comment (jonathanlastmileai) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/__init__.py:5]: This errors for me FYI

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py
Comment (jonathanlastmileai) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:32]: P2

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py
Comment (jonathanlastmileai) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:101]: it's a TON of duplicated and confusing code...there's no way I could have ton this reasonably quickly if it wasn't almost identical to generation

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py
Comment (jonathanlastmileai) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:242]: summarize expects a positional argument "inputs" which differs from gen API

Comment (rossdanlm) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:242]: Oh cool!

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/__init__.py
Comment (rossdanlm) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/__init__.py:5]: Huh weird, ok can you pls file issue for this and assign to me to followup?

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py
Comment (jonathanlastmileai) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:302]: I think it's better to fix afterward, but ideally we wouldn't duplicate so much to begin with for this kind of reason

File: cookbooks/Getting-Started/travel.aiconfig.json
Comment (Ankush-lastmile) [inline cookbooks/Getting-Started/travel.aiconfig.json:7]: please remove changes made to the getting started cookbook

**Review by Ankush-lastmile — DISMISSED** [review]
requesting changes to make sure that the cookbook changes do not get merged

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py
Comment (jonathanlastmileai) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:140]: thanks, I'll remove these

**Comment by jonathanlastmileai** [comment]
@rossdanlm which ones are just nits ? Are we good to go?

@Ankush-lastmile

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py
Comment (Ankush-lastmile) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py:299]: why does it need to be fixed afterwards? it looks like a one line change here

**Comment by Ankush-lastmile** [comment]
> Are we good to go?

I think you have a couple of comments left unresolved, I removed my request for changes. Thanks for removing changes to the cookbook.

**Pull request merged**
